src/helpers/promises.ts

```diff
@@ -37,3 +37,106 @@ export const resolveAfterPending = (): Promise<void> => {
 export const resolveNextTick = (): Promise<void> => {
   return new Promise((resolve) => setTimeout(resolve));
 };
+
+/**
+ * A class to manage timing functions and arbitrary periods of time before
+ * generating a `Server-Timing` header for use in HTTP responses.
+ *
+ * This is a very simple implementation that does not support nested timings or
+ * fractions of a millisecond.
+ */
+export class ServerTiming {
```

Comment:
Could be nice as it's own file, eventually with it's own tests - feels like an standalone functionality that's not just a promise helper 🤔

Reply:
@djfarrelly Great shout!